### ISSUE TITLE
Stacking-order z-index: 0 bug

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#4542]
There's a bug in our [getStackLevel](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/dom/create-grid.js#L313) code where we treat a declared `z-index: 0` as having a stacking context but not a z-index. 

For example, the correct stack order of the following code should be `[#parent, #child, body, html]` but instead it comes out to `[#parent, body, html, #child]`

```html
<div id="parent" style="position: relative: z-index: 0">
  <div id="child" style="position: relative; z-index: -1; background: red"></div>
</div>
```

Instead we need to allow `z-index: 0` to be treated as a valid `z-index` value and allow the stacking level to equal 0.  However doing so creates another problem though as we made the `ROOT_LEVEL` z-index equal 0 and then remove [root level elements when we encounter a real stack](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/dom/create-grid.js#L274). (a positioned stacking context that declares a non-auto z-index). 

We'll also need to update our `visuallySort` function as it treats the -1 z-index of a child context as being higher in the visual stack than the parent element, when in reality the parent should be on top.

**Comment by padmavemulapati** [comment]
Verified with the latest axe-core develop code base, `color-contrast` violation is not going into false negative and correctly determines that one of the ancestor container elements is rendering on top of the leaf descendant element

not in `incomplete` but in `violations` with proper stack order

https://github.com/user-attachments/assets/d5c31f0c-1a23-4107-bcd6-5552d226d38a

as well, with the code snippet:

```

<div id="1" style="position: relative; z-index: 4; width: 200px; height: 200px;">
          <div id="2" style="position: absolute;">
            <div id="3" style="transform: translate3d(0px, 0px, 0px);">
              <span id="target">text</span>
            </div>
          </div>
        </div>
  
        <div style="position: relative; z-index: 1; width: 200px; height: 200px;">
  <div style="position: absolute;">
    <div style="transform: translate3d(0px, 0px, 0px);">
      <span id="fail13" style="background-color: #111; color: #000">low contrast</span>
      <span id="pass24" style="background-color: #fff; color: #000">ok contrast</span>
    </div>
  </div>
</div>

<div style="position: relative; z-index: 1; width: 200px; height: 200px;">
  <div style="position: absolute;">
    <div style="opacity: 0.99;">
      <span id="fail14" style="background-color: #111; color: #000">low contrast</span>
      <span id="pass25" style="background-color: #fff; color: #000">ok contrast</span>
    </div>
  </div>
</div>

<body style="background-color: #121212; color: #333">
  <div style="position: fixed; width: 100%; height: 100vh;">
    <svg style="position: absolute">
      <g>
        <g transform="translate(212.26931762695312, 0) scale(1)">
          <path stroke="#9e72c9" fill="#7435b2" d=" m 0,0  h 15  l 6,4  3,0  6,-4  h 133.06036376953125  v 5  V 5  V 29  V 29  V 34  h -133.06036376953125  l -6,4  -3,0  -6,-4  h -15  H 0 z
            "></path>
          <g transform="translate(10,9.5)">
            <text fill="#fff" dominant-baseline="central" x="0" y="7.5">play</text>
          </g>
        </g>
      </g>
    </svg>
  </div>
</body>
<style>
  #container {
    position: fixed;
    background-color: #333;
    padding: 0.5em 2em;
  }
  #target {
    filter: contrast(100%);
    color: #eee;
  }
</style>
<div id="container">
  <span id="target"> Hello world </span>
</div>
 
</html>
```


<img width="1329" height="1002" alt="Image" src="https://github.com/user-attachments/assets/438707e3-e1bb-4be1-987a-acb77f3cc6c1" />
